soc/nordic/CMakeLists.txt

```diff
@@ -17,7 +17,11 @@ zephyr_library_sources(
 # headers for different SoCs.
 set(dt_binding_includes ${DTS_INCLUDE_FILES})
 list(FILTER dt_binding_includes INCLUDE REGEX "/dt-bindings/.*\.h$")
-list(TRANSFORM dt_binding_includes PREPEND "-include;")
+if(CMAKE_C_COMPILER_ID STREQUAL "IAR")
```

Comment:
we should avoid compiler / toolchain specific handling inside the code as it makes support for multiple toolchains much harder.

Instead the code itself should be cleaned up and the include flag should be defined in the compiler properties and picked up where used.

For example like this:

```
set(include_flag $<TARGET_PROPERTY:compiler,include-file>)

set_source_files_properties(
  validate_binding_headers.c
  DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
  PROPERTIES COMPILE_OPTIONS "${include_flag}$<JOIN:${dt_binding_includes},;${include_flag}>"
)
```

and then in https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/compiler/gcc/compiler_flags.cmake add a line like this:
```
set_compiler_property(PROPERTY include-file "-include$<SEMICOLON>")
# Note to @RobinKastberg: check if "-include;" is sufficient in above.
```

and in the correspong IAR `compiler_flags.cmake` you can then do similar:
```
set_compiler_property(PROPERTY include-file "-preinclude$<SEMICOLON>")
```
and there will no longer be any need for doing compiler check in the CMake code itself, plus a benefit is that another toolchain can just provide its own flag and doesn't have to add an extra if-clause to this CMake code.